src/debugger/debugSessionBase.ts

```diff
@@ -104,10 +117,34 @@ export abstract class DebugSessionBase extends LoggingDebugSession {
         }
     }
 
-    protected showError(message: string, response: DebugProtocol.Response): void {
+    protected async disconnectRequest(response: DebugProtocol.DisconnectResponse, args: DebugProtocol.DisconnectArguments, request?: DebugProtocol.Request): Promise<void> {
+        await this.appLauncher.getRnCdpProxy().stopServer();
+
+        this.cancellationTokenSource.cancel();
+        this.cancellationTokenSource.dispose();
+
+        // Then we tell the extension to stop monitoring the logcat, and then we disconnect the debugging session
+        if (this.previousAttachArgs.platform === "android") {
+            try {
+                this.appLauncher.stopMonitoringLogCat();
+            } catch (err) {
+                logger.warn(localize("CouldNotStopMonitoringLogcat", "Couldn't stop monitoring logcat: {0}", err.message || err));
+            }
+        }
+
+        super.disconnectRequest(response, args, request);
+    }
+
+    protected showError(error: Error, response: DebugProtocol.Response): void {
+        if ((error instanceof InternalError || error instanceof NestedError)
+            && error.errorCode === InternalErrorCode.CancellationTokenTriggered
```

Comment:
Do we need to completely stop showing errors if session is stopped? We may miss some important information for debugging some bugs if we do this. What do you think? Or maybe I am missing something?